### PR TITLE
Manual backport: datetime-diff function for SQL Server #27002

### PR DESCRIPTION
**Pull request by calherries**
This is a manual backport for https://github.com/metabase/metabase/pull/27002

### CODE REVIEW

File: test/metabase/query_processor_test/date_time_zone_functions_test.clj [hunk 1]
```diff
@@ -831,9 +839,8 @@
                       first))))))))
 
 (deftest datetime-diff-type-test
-  ;; FIXME — The excluded drivers below don't have TIME types. These shouldn't be hard-coded with #26807
-  (mt/test-drivers (disj (mt/normal-drivers-with-feature :datetime-diff)
-                         :oracle :presto :redshift :sparksql :snowflake)
+  (mt/test-drivers (->> (disj (mt/normal-drivers-with-feature :datetime-diff) :snowflake)
+                        (filter mt/supports-time-type?))
```
Comment (calherries) [inline test/metabase/query_processor_test/date_time_zone_functions_test.clj:843]: Note snowflake coverage on this test will be added here https://github.com/metabase/metabase/pull/27068#top

File: test/metabase/query_processor_test/date_time_zone_functions_test.clj [hunk 2]
```diff
@@ -734,7 +734,9 @@
                                  "2022-10-03T00:00:00Z")))))      ; 2022-10-03T00:00:00Z
         (testing "hour under a day"
           (mt/with-temporary-setting-values [driver/report-timezone "Atlantic/Cape_Verde"]
-            (is (partial= {:second 82800 :minute 1380 :hour 23 :day 1}
+            (is (partial= (if (driver/supports? driver/*driver* :set-timezone)
+                            {:second 82800 :minute 1380 :hour 23 :day 1}
+                            {:second 82800 :minute 1380 :hour 23 :day 0})
```
Comment (calherries) [inline test/metabase/query_processor_test/date_time_zone_functions_test.clj:739]: SQL server doesn't support set-timezone, so this change verifies that setting the report-timezone doesn't have an effect.

**Review by camsaul — APPROVED** [review]
I'll go ahead and 👍 since this is just a backport but I would have liked to have reviewed the original version of it

**Review by camsaul — CHANGES_REQUESTED** [review]
Sorry, I think I jumped the gun on approving this.

Our general policy is not to ship new features in point releases. Point releases are just for bug and security fixes. It seems unlikely, but it's possible this change can introduce new bugs, so it's best to give it bake time.

Maybe there is some really compelling reason we would want to ship this in a point release, but I haven't heard anything about doing so. Let me know if there is actually some reason to ship this in 45.x

**Comment by calherries** [comment]
My bad, I was under the impression that adding driver support for existing custom expression functions was something we could do in a point release. Nobody I've talked to about this had suggested otherwise.

For transparency snowflake's datetimeDiff support was added in 0.45.1 https://github.com/metabase/metabase/pull/26746
and redshift and vertica support is geared up to go in 0.45.2. I'll revert redshift and vertica backports.

**Pull request closed without merging**